src/Core/Model/Common/JsonObject.php

```diff
@@ -175,7 +175,7 @@ protected function initialize($field)
              * @var JsonDeserializeInterface $type
              */
             $value = $this->getRaw($field, null);
-            if (!is_null($value)) {
+            if (!is_null($value) && is_array($value)) {
```

Comment:
Out of curiosity. Do you have an example when this could happen?

Reply:
When trying to update our app to php 7.4, there are problems adding items to the shopping cart and flushing the shopping cart. With certain custom fields (date value), the function "$type::fromArray" throws the error message that the given "value" is not an array.

The is_array function can at least ensure that the application does not crash.

I would like to know how the integration tests should work. I checked out the repository and entered Commerctools data. Nevertheless, the integration tests throws over 3000 errors.

Reply:
I see. Will take a look if I can create a reproducing test.

The Integration tests need either these env variables set:

- `COMMERCETOOLS_CLIENT_ID`
- `COMMERCETOOLS_CLIENT_SECRET`
- `COMMERCETOOLS_PROJECT`

or you have to copy the `tests/integration/myapp.yml.dist` to `tests/integration/myapp.yml` and fill with the credentials. The needed scopes for your API client you can find here: 

https://github.com/commercetools/commercetools-php-sdk#running-integration-tests

Reply:
I ran the composer install locally. I was asked for CT data(COMMERCETOOLS_CLIENT_ID etc), I entered the data and the tests\myapp.yml file was successfully created. Required scopes are set!
no code changes and the result is:
....
45) Commercetools\Core\IntegrationTests\Errors\ErrorResponseTest::testInsufficientScope
Failed asserting that 400 is equal to expected exception code 403.

46) Commercetools\Core\IntegrationTests\Me\MeCartRequestTest::testCustomerNoAccess
Failed asserting that Commercetools\Core\Error\ApiError Object (...) is an instance of class "Commercetools\Core\Error\ResourceNotFoundError".

C:\workspace\commercetools-php-sdk\tests\integration\Me\MeCartRequestTest.php:379

47) Commercetools\Core\IntegrationTests\Me\MeCartRequestTest::testAnonNoCart
Failed asserting that null is an instance of class "Commercetools\Core\Error\ResourceNotFoundError".

C:\workspace\commercetools-php-sdk\tests\integration\Me\MeCartRequestTest.php:523

48) Commercetools\Core\IntegrationTests\Me\MeCartRequestTest::testCustomerNoCart
Failed asserting that null is an instance of class "Commercetools\Core\Error\ResourceNotFoundError".

C:\workspace\commercetools-php-sdk\tests\integration\Me\MeCartRequestTest.php:554

49) Commercetools\Core\IntegrationTests\Me\MeOrderRequestTest::testCustomerNoAccess
Failed asserting that Commercetools\Core\Error\ApiError Object (...) is an instance of class "Commercetools\Core\Error\ResourceNotFoundError".

C:\workspace\commercetools-php-sdk\tests\integration\Me\MeOrderRequestTest.php:216

Reply:
Strange. Makes so no sense 🤷‍♂️

Btw. could you give me some example data for your problem?